2021/day10/day10.java

```diff
@@ -0,0 +1,98 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+
+public class day10 {
+    public static class Scores {
+        int totalSyntaxScore;
+        Long middleScore;
+
+        public Scores(int x, Long y) {
+            totalSyntaxScore = x;
+            middleScore = y;
+        }
+    }
+
+    public static ArrayList<ArrayList<Character>> readInput(String filename) {
+        ArrayList<ArrayList<Character>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Character> line = new ArrayList<>();
+                String s = sc.next();
+                for (int i = 0; i < s.length(); i++) {
+                    line.add(s.charAt(i));
+                }
+                input.add(line);
+            }
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    private static final HashMap<Character, Integer> openingScores = new HashMap<>();
+    static {
+        openingScores.put('(', 1);
+        openingScores.put('[', 2);
+        openingScores.put('{', 3);
+        openingScores.put('<', 4);
+    }
+
+    private static final HashMap<Character, Integer> closingScores = new HashMap<>();
+    static {
+        closingScores.put(')', 3);
+        closingScores.put(']', 57);
+        closingScores.put('}', 1197);
+        closingScores.put('>', 25137);
+    }
+
+    public static Scores syntaxScoring() {
+        ArrayList<ArrayList<Character>> input = readInput("data.txt");
+        ArrayList<Long> autocompleteScores = new ArrayList<>();
+        ArrayList<Character> openingChars =
+        new ArrayList<Character>(Arrays.asList('<', '(', '[', '{'));
+        ArrayList<Character> closingChars =
+        new ArrayList<Character>(Arrays.asList('>', ')', ']', '}'));
```

Comment:
Weird indentation, either tabulate this or put `new ArrayList<Character>(` on line above and the rest (tabulated) here, so that it's clear that it's a continuation of the last line